src/app/loadout-drawer/loadout-utils.ts

```diff
@@ -27,21 +29,43 @@ const gearSlotOrder: DimItem['type'][] = [
 /**
  * Creates a new loadout, with all of the items equipped and the items inserted mods saved.
  */
-export function newLoadout(name: string, items: LoadoutItem[], modsHashes?: number[]): Loadout {
+export function newLoadout(name: string, items: LoadoutItem[]): Loadout {
   return {
     id: uuidv4(),
     classType: DestinyClass.Unknown,
     name,
     items,
-    parameters: modsHashes?.length
-      ? {
-          mods: modsHashes,
-        }
-      : undefined,
     clearSpace: false,
   };
 }
 
+/**
+ * Create a new loadout that includes all the equipped items and mods on the character.
+ */
+export function newLoadoutFromEquipped(name: string, dimStore: DimStore) {
+  const items = dimStore.items.filter(
+    (item) => item.equipped && itemCanBeInLoadout(item) && fromEquippedTypes.includes(item.type)
+  );
+  const loadout = newLoadout(
+    name,
+    items.map((i) => {
+      const item = convertToLoadoutItem(i, true);
+      if (i.bucket.hash === BucketHashes.Subclass) {
```

Comment:
I've been using `bucket.type === 'Class'` for a lot of this as its what is used in `LoadoutDrawerContents`. Is it worth syncing them all or should they be the same thing?

Reply:
I like the hashes version in general - both are all over.